client/app/collections/scheduleitems.coffee

```diff
@@ -3,19 +3,27 @@ module.exports = class ScheduleItemsCollection extends Backbone.Collection
     model: require '../models/scheduleitem'
     comparator: (si1, si2) -> si1.getDateObject().diff si2.getDateObject()
 
+    # The calendar tag for all event was referencing the same instance
+    # in every event object. This is not the case anymore, so we have
+    # to check every event calendar in the a given calendars collection
+    # to verify the visibility
+    visibleItems: (calendars) ->
+        return _ @filter (item) ->
```

Comment:
Why the `_` here? (plus, explicit `return` isn't really needed, you can safely drop it)